pkg/agent/manager/cache/cache.go

```diff
@@ -375,6 +420,40 @@ func (c *Cache) UpdateEntries(update *UpdateEntries, checkSVID func(*common.Regi
 		}
 	}
 
+	// entries with active subscribers which are not cached will be put in staleEntries map
+	activeSubsByEntryID, recordsWithLastAccessTime := c.syncSVIDs()
+	extraSize := len(c.svids) - c.svidCacheMaxSize
+
+	// delete svids without subscribers and which have not been accessed since svidCacheExpiryTime
+	if extraSize > 0 {
```

Comment:
The entire `UpdateEntries` method is ripe for refactoring:) 
I've added more comments let me know if you still think it would still be better to kick deletion in separate method.